unattended.sh

```diff
@@ -1,186 +1,766 @@
 #!/bin/sh
 
-function error_and_exit {
-  echo -e "$1"
-  exit 1
-}
+### Define all supported constants
+OPTIONS=":t:v:r:l:"
+declare -A SUPPORTED_LOG_LEVEL=([DEBUG]=0 [INFO]=1 [WARN]=2 [ERROR]=3)
+declare -A SUPPORTED_TOPOLOGY=([central]=1 [poller]=1)
+declare -A SUPPORTED_VERSION=([21.04]=1)
+declare -A SUPPORTED_REPOSITORY=([testing]=1 [unstable]=1 [stable]=1)
+default_timeout_in_sec=5
+script_short_name="$(basename $0)"
+default_ip=$(hostname -I | awk '{print $1}')
+###
 
-function print_step_begin {
-  echo -e "\n$1..."
-}
+#Define default values
+topology=${ENV_CENTREON_TOPOLOGY:-"central"}   #Default topology to be installed
+version=${ENV_CENTREON_VERSION:-"21.04"}       #Default version to be installed
+repo=${ENV_CENTREON_REPO:-"stable"}            #Default repository to used
+operation=${ENV_CENTREON_OPERATION:-"install"} #Default operation to be executed
+runtime_log_level=${ENV_LOG_LEVEL:-"INFO"}     #Default log level to be used
+selinux_mode=${ENV_SELINUX_MODE:-"permissive"} #Default SELinux mode to be used
+wizard_autoplay=${ENV_WIZARD_AUTOPLAY:-"true"} #Default the install wizard is run auto
+central_ip=${ENV_CENTRAL_IP:-$default_ip}      #Default central ip is the first of hostname -I
+
+#Generate random MariaDB root password
+mariadb_root_password=$(
+	date +%s%N | sha256sum | base64 | head -c 32
+	echo
+)
+
+#Generate random Centreon admin password
+centreon_admin_password=$(
+	date +%s%N | sha256sum | base64 | head -c 32
+	echo
+)
+
+#Generate random MariaDB centreon password
+mariadb_centreon_password=$(
+	date +%s%N | sha256sum | base64 | head -c 32
+	echo
+)
```

Comment:
```suggestion
# Set MariaDB password from ENV or random password if not defined
mariadb_root_password=${ENV_MARIADB_ROOT_PASSWD:-"$(genpasswd)"}

# Set from ENV or random Centreon admin password
centreon_admin_password=${ENV_ADMIN_PASSWD:-"$(genpasswd)"}

# Set from ENV or random MariaDB centreon password
mariadb_centreon_password=${ENV_CENTREON_PASSWD:-"$(genpasswd)"}

# Set from ENV or Administrator first name
centreon_admin_firstname=${ENV_ADMIN_USERNAME:-"Jonh"}

# Set from ENV or Administrator last name
centreon_admin_lastname=${ENV_ADMIN_USERNAME:-"Don"}

# Set from ENV or Administrator e-mail
centreon_admin_email=${ENV_ADMIN_EMAIL:-"admin@admin.tld"}
```

Add possibility to pass the password and other fields of the wizard by environment variable